pkg/apis/core/v1/helper/helpers.go

```diff
@@ -85,6 +85,21 @@ func HugePageSizeFromResourceName(name v1.ResourceName) (resource.Quantity, erro
 	return resource.ParseQuantity(pageSize)
 }
 
+// IsHugePageMedium returns true if the volume medium is in 'HugePages[-size]' format
+func IsHugePageMedium(medium v1.StorageMedium) bool {
+	return medium == v1.StorageMediumHugePages || strings.HasPrefix(string(medium), string(v1.StorageMediumHugePagesPrefix))
+}
+
+// HugePageSizeFromMedium returns the page size for the specified huge page medium.
+// If the specified input is not a valid huge page medium an error is returned.
+func HugePageSizeFromMedium(medium v1.StorageMedium) (resource.Quantity, error) {
```

Comment:
Hmm. I feel like this naming scheme would result in some ambiguity, since this code will not work, even tho my intuition is that is should

```go
medium := "HugePages"
if isHugePageMedium := IsHugePageMedium(medium); isHugePageMedium {
    size, err := HugePageSizeFromMedium(medium)
    // err will not be nil here
    [...]
}

```

Reply:
@odinuge  I don't see any ambiguity here. If the medium is valid huge page medium it doesn't mean that we can get huge page size from it.

Reply:
Hmm.

Guess we can add another check `if medium == v1.StorageMediumHugePages` and return an error saying that we cannot get the size from it, since it will now fail on `ParseQuantity`

Reply:
yes, that makes sense to me. will do.

Reply:
done